pkg/agent/handler/accessrequestdefinition.go

```diff
@@ -0,0 +1,36 @@
+package handler
+
+import (
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+type ardHandler struct {
+	agentCacheManager agentcache.Manager
+}
+
+// NewARDHandler creates a Handler for Access Requests
+func NewARDHandler(agentCacheManager agentcache.Manager) Handler {
+	return &ardHandler{
+		agentCacheManager: agentCacheManager,
+	}
+}
+
+// Handle processes grpc events triggered for AccessRequests
+func (h *ardHandler) Handle(action proto.Event_Type, meta *proto.EventMeta, resource *v1.ResourceInstance) error {
+	if resource.Kind != mv1.AccessRequestDefinitionGVK().Kind {
+		return nil
+	}
+
+	if action == proto.Event_CREATED || action == proto.Event_UPDATED {
+		h.agentCacheManager.AddAccessRequestDefinition(resource)
```

Comment:
if this can return error, it would get eaten up by the return nil, right?

Reply:
@tsjohns9?

Reply:
Looks like the add funcs in the cache manager do not return error

Reply:
coolio juliio